contributors/devel/sig-testing/e2e-tests.md

```diff
@@ -383,8 +383,10 @@ one. To start a local cluster, you can run:
 
 ```sh
 # The PATH construction is needed because PATH is one of the special-cased
-# environment variables not passed by sudo -E
-sudo PATH=$PATH hack/local-up-cluster.sh
+# environment variables not passed by sudo -E.
+# USER is need for the script to change ownership of client admin key to
+# current user. Otherwise, only root can access the cluster.
+sudo PATH=$PATH USER=$(id -u) hack/local-up-cluster.sh
```

Comment:
`local-up-cluster.sh` changes owner of client admin key file to `USER` [1], which is supposed to be current non-root user. But we are running the command with `sudo`, the USER will always to root [2], and the current user cannot read client admin key to access the deployed cluster.

[1] https://github.com/kubernetes/kubernetes/blob/master/hack/local-up-cluster.sh#L628
[2] https://github.com/kubernetes/kubernetes/blob/master/hack/local-up-cluster.sh#L57

Reply:
I'd prefer to defer to @BenTheElder and @dims here as well as I don't have much experience with `hack/local-up-cluster.sh`

Reply:
Agree let's go with this @cyb70289 
```
[dims@dims-a01 21:08] ~/go/src/k8s.io/kubernetes/hack ⟩ rg USER local-up-cluster.sh
57:USER=${USER:-$(whoami)}
628:    ${CONTROLPLANE_SUDO} chown "${USER}" "${CERT_DIR}/client-admin.key" # make readable for kubectl
```

Reply:
I sort of thing it should be owned by root given we needed root to run this?
Because this kubeconfig has control over the container runtime on the host outright and is tantamount to being root.

You could make a similar argument about KIND, but it is not such a straightforward control, and if the user has docker socket access limited to a particular user they will also need to run kind as that user and similarly not own the file.

Reply:
Agreed this is a security issue. I've reverted the change.